### ISSUE TITLE
VertexSizeComparator is a pure pile of poo

### ISSUE THREAD

**Issue by Riduidel** [Riduidel/gaedo#90]
The lame coder who wrote that (me) wrote a shameless

```
@Override
public int compare(VertexSet o1, VertexSet o2) {
    int returned = (int) Math.signum(o1.size()-o2.size());
    if(returned==0) {

    }
    return returned;
}
```

And guess what ? When using #88 (which is a huge step forward), this poor implementation covers me with shame, developpers hatred, and all other unpleasant things.

**Comment by Riduidel** [comment]
Seems like I did exactly this : http://lesjoiesducode.fr/post/88376624833/quand-je-corrige-un-bug-et-que-ca-en-cree-un-nouveau

More exactly, the new code makes more use of `VertexRootsCollector` (because it uses more input `VertexSet` and links them to `VertexTest` instances). If you take a look at  `VertexRootsCollector#getSetsToProcessedTests(...)`, you'll see it outputs a `SortedMap<VertexSet, VertexTest>` in which the key (the **best** `VertexSet`) is used to choose the `VertexTest` to apply. But, due to the poor implementation of `VertexSetSizeComparator`, it is possible to have two `VertexSet`be considered as one and, as a consequence, have one `VertexTest` linked to the wrong `VertexSet`

In other words, considering the `TestFor90_QueriesWithAndAreMadeUnusable` class, we expect to ahve three sets to tests mappings, but only one exists in map (because all sets have the same size). Furthermore, due to [`Map#put`](http://docs.oracle.com/javase/7/docs/api/java/util/Map.html#put%28K,%20V%29) semantics, when one new `VertexSet` is pushed over the existing one, the existing key is not changed, but the value is, which makes us use an incorrect `VertexTest` for the given `VertexSet`.

Hope this is clear enough for you ... for me, it is.
